cmd/cstor-pool-mgmt/controller/backup-controller/new_backup_controller.go

```diff
@@ -0,0 +1,246 @@
+			bkp := obj.(*apis.BackupCStor)
+			if !IsRightCStorPoolMgmt(bkp) {
+				return
+			}
+			glog.Infof("BackupCStor Resource delete event: %v, %v", bkp.ObjectMeta.Name, string(bkp.ObjectMeta.UID))
+		},
+	})
+	return controller
+}
+
+// enqueueBackupCStor takes a BackupCStor resource and converts it into a namespace/name
+// string which is then put onto the work queue. This method should *not* be
+// passed resources of any type other than BackupCStor.
+func (c *BackupController) enqueueBackupCStor(obj *apis.BackupCStor, q common.QueueLoad) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(obj); err != nil {
+		runtime.HandleError(err)
+		return
+	}
+	q.Key = key
+	c.workqueue.AddRateLimited(q)
+}
+
+// cleanupOldBackup set fail status to old pending backup
+func (c *BackupController) cleanupOldBackup(clientset clientset.Interface) {
+	bkplabel := "cstorpool.openebs.io/uid=" + os.Getenv(string(common.OpenEBSIOCStorID))
+	bkplistop := metav1.ListOptions{
+		LabelSelector: bkplabel,
+	}
+	bkplist, err := clientset.OpenebsV1alpha1().BackupCStors(metav1.NamespaceAll).List(bkplistop)
+	if err != nil {
+		return
+	}
+
+	for _, bkp := range bkplist.Items {
+		switch bkp.Status {
+		case apis.BKPCStorStatusInProgress:
+			//Backup was in in-progress state
+			laststat := findLastBackupStat(clientset, bkp)
+			updateBackupStatus(clientset, bkp, laststat)
+		case apis.BKPCStorStatusDone:
+			continue
+		default:
+			//Set backup status as failed
+			updateBackupStatus(clientset, bkp, apis.BKPCStorStatusFailed)
+		}
+	}
+}
+
+// updateBackupStatus will update the backup status to given status
+func updateBackupStatus(clientset clientset.Interface, bkp apis.BackupCStor, status apis.BackupCStorStatus) {
+	bkp.Status = status
+
+	_, err := clientset.OpenebsV1alpha1().BackupCStors(bkp.Namespace).Update(&bkp)
+	if err != nil {
+		glog.Errorf("Failed to update backup(%s) status(%s)", status, bkp.Name)
+		return
+	}
+}
+
+// findLastBackupStat will find the status of backup from last-backup
+func findLastBackupStat(clientset clientset.Interface, bkp apis.BackupCStor) apis.BackupCStorStatus {
+	lastbkpname := bkp.Spec.BackupName + "-" + bkp.Spec.VolumeName
+	lastbkp, err := clientset.OpenebsV1alpha1().BackupCStorLasts(bkp.Namespace).Get(lastbkpname, v1.GetOptions{})
+	if err != nil {
+		// Unable to fetch the last backup, so we will return fail state
+		glog.Errorf("Failed to fetch last backup:%s error:%s", lastbkpname, err.Error())
+		return apis.BKPCStorStatusFailed
+	}
+
+	// let's check if snapname matches with current snapshot name
+	if bkp.Spec.SnapName == lastbkp.Spec.SnapName || bkp.Spec.SnapName == lastbkp.Spec.PrevSnapName {
+		return apis.BKPCStorStatusDone
+	}
+
+	// lastbackup snap/prevsnap doesn't match with bkp snapname
+	return apis.BKPCStorStatusFailed
+}
+
+// handleBKPAddEvent is to handle add operation of backup controller
+func (c *BackupController) handleBKPAddEvent(bkp *apis.BackupCStor, q *common.QueueLoad) {
```

Comment:
There are so many method names with event & handle.
It is very confusing.
Can we provide appropriate method names or probably combine them if that makes more sense.